src/ParseQuery.js

```diff
@@ -1028,11 +1030,22 @@ class ParseQuery {
    *     iteration has completed.
    */
   async reduce(callback: (accumulator: any, currentObject: ParseObject, index: number) => any, initialValue: any, options?: BatchOptions): Promise<Array<any>> {
-    const objects = [];
+    let accumulator = initialValue;
+    let index = 0;
     await this.each((object) => {
-      objects.push(object);
+      // If no initial value was given, we take the first object from the query
+      // as the initial value and don't call the callback with it.
+      if (index === 0 && initialValue === undefined) {
```

Comment:
[Calling reduce() on an empty array without an initialValue will throw a TypeError.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce)

should we handle it the same way?

Reply:
Good point! Updated to have similar behavior in 687e795. Thanks :smile: